docs/how-to/hip_runtime_api/opengl_interop.rst

```diff
@@ -0,0 +1,66 @@
+.. meta::
+   :description: HIP provides an OpenGL interoperability API that allows
+                 efficient data sharing between HIP's computing power and
+                 OpenGL's graphics rendering.
+   :keywords: AMD, ROCm, HIP, OpenGL, interop, interoperability
+
+********************************************************************************
+OpenGL interoperability
+********************************************************************************
+
+The HIP--OpenGL interoperation involves mapping OpenGL resources, such as
+buffers and textures, for HIP access. This mapping process enables HIP to
+utilize these resources directly, bypassing the need for data transfers between
+the CPU and GPU. This capability is useful in applications that require both
+intensive GPU computation and real-time visualization.
+
+The OpenGL resource can be mapped to HIP with :cpp:func:`hipGraphicsMapResources`
```

Comment:
I feel like these functions deserve a little more description as to how the user should use them.  For instance, the reference material provides little explanation about the various parameters to the function. What does "Stream for synchronization" refer to in the hipGraphicsMapResources() function? How does one determine the "Size of the buffer accessible from devPtr" for the hipGraphicsResourceGetMappedPointer() function?

Reply:
I updated the description here, but I would not add the argument description. "Stream for synchronization" should be described in the section where the stream is introduced while the size of the buffer should be known by the developer, before using the map function.

Reply:
Is the size of the buffer static, or varies depending on what the developer is trying to do? If the size is static according to the mapped buffer, then why does the user need to specify it? If the size of the buffer depends on the users needs, then how do they ensure the requested size does not exceed the mapped size?

Reply:
The OpenGL buffer size not static, and I checked we can get the buffer size with the following function call:

```
GLint size = 0;

glGetBufferParameteriv(GL_ARRAY_BUFFER, GL_BUFFER_SIZE, &size);
```

The hipGraphicsMapResources function returns error codes (hipSuccess, hipErrorInvalidValue, hipErrorUnknown or hipErrorInvalidResourceHandle), but I am not sure, if it's checking the buffer size. If the size of the buffer invalid, we may get a HIP error during the mapping, or we will get into trouble, while we try to use the mapped resource. In C and C++ the invalid memory access can happen.

"I feel like these functions deserve a little more description"

I wanted to add more here, but simple there is not much more I can write about this function. I can add a note for glGetBufferParameteriv function, if it helps.

Reply:
Thanks for checking. I think I will touch base with Julia as some of this information should appear in the API reference.